### DISCUSSION TITLE
How do i use Internal DHCP server in a single NIC isolated network

### DISCUSSION THREAD

**Discussion by pzerone** [ltsp/ltsp#899] [Q&A] [answered]
I have an LTSP deployment with a single NIC setup and proxy the external DHCP server for ip addresses.
How would i convert it so that the server is connected to an isolated switch without an external DHCP server, using a single NIC. Couple of clients are connected to the switch, booting from the LTSP server and getting ip addresses from dnsmasq of the LTSP server.
What are the configs i should change from a default installation of LTSP with a single NIC setup. I figured i should change the `ltsp-dnsmasq.con` so that it starts the dhcp server instead of proxying them. anything else i should adjust?.
I'm confused about the `real dhcp server` part and if it will conflict with my requirement.

**Comment by alkisg** [comment]
The easiest way is to set the server IP to 192.168.67.1 and run `sudo ltsp dnsmasq`.

If you want a different subnet, create /etc/dnsmasq.d/local.conf with the appropriate range, and run `sudo ltsp dnsmasq -p0` which means no proxy server.

**Comment by pzerone** [comment]
`sudo ltsp dnsmasq` will try to use the external DHCP server right? in my case, there won't be an external DHCP server or any other server for that matter, it will an isolated, fully offline network.
Ideally, i would like to use my own subnet, and lease from that range. so i think the second option might work, but isn't that the same command for 2 NIC setup? which will check if the ip is `192.168.67.1` before starting the dhcp server?.

and additionally, can i make the `[proxy/no-proxy]` toggle-able without a server reboot?(client reboots are okay).

**Comment by alkisg** [comment, marked as answer]
A plain `sudo ltsp dnsmasq` activates the real DHCP only on the 192.168.67.0/24 subnet. That's why it'll work if and only if your server IP is in that subnet.

Single or 2 NIC setup makes no difference; the ` dhcp-range=start,end,lease-time` syntax checks IPs, not interfaces.
So if you create local.conf with a line like `dhcp-range=10.0.0.20,10.0.0.250,12h` and you restart dnsmasq, then you have a DHCP server in that subnet.

You never need to reboot the server for dnsmasq changes. Just run `sudo systemctl restart dnsmasq` whenever you manually change the dnsmasq config (the `ltsp dnsmasq` command does that for you).

**Comment by pzerone** [comment]
Got it, so here is how I understand this:

1) if we need internal DHCP, either do just `ltsp dnsmasq -p0` for internal DHCP from the 192.168.67.0/24 subnet or set custom DHCP range on `local.conf` file and then run the same command for DHCP from our custom subnet.

2) If we need to proxy DHCP, make sure to have an external DHCP server on the network and the ltsp server recieved it's IP from it, and then run the plain ol `ltsp dnsmasq`.

3) we can toggle between these two setups as long as these conditions are met (having the ext DHCP server/ having the custom lease range in `local.conf` file, etc)

Did I understand it correctly?

And also, `real-dhcp` means internal and `proxy-dhcp` means external right?

**Comment by alkisg** [comment]
Yes you understood correctly.
real-dhcp means the LTSP server will give the IPs, while proxy-dhcp means another server will give the IPs, and LTSP will only give the boot filename.

**Comment by pzerone** [comment]
Got it. Thank you so much for the help.
